core/src/main/java/com/datastax/oss/driver/internal/core/config/typesafe/DefaultProgrammaticDriverConfigLoaderBuilder.java

```diff
@@ -62,6 +83,18 @@ public DefaultProgrammaticDriverConfigLoaderBuilder() {
     this(DEFAULT_FALLBACK_SUPPLIER, DefaultDriverConfigLoader.DEFAULT_ROOT_PATH);
   }
 
+  /**
+   * An instance of a programmatic builder that uses application specific config defaults from the
+   * specified application CLassLoader to override Driver config defaults for config that is not
+   * supplied programmatically.
+   *
+   * @param appClassLoader The application's ClassLoader from which to load application
+   *     configuration defaults.
+   */
+  public DefaultProgrammaticDriverConfigLoaderBuilder(ClassLoader appClassLoader) {
+    this(getDefaultConfigSupplier(appClassLoader), DefaultDriverConfigLoader.DEFAULT_ROOT_PATH);
+  }
```

Comment:
There wasn't an easy way to automatically get a hold of an OSGi bundle's ClassLoader from within the Driver's code as the Driver is a separate bundle and doesn't know which bundle's might be using it. For this to work in an OSGi container, an application bundle using the programmatic config loader will have to pass in a reference to its ClassLoader here. This allows us to create a Config supplier that can locate application bundled resources, instead of letting TypeSafe try to use the context ClassLoader, which isn't the same class loader as the OSGi app bundle ClassLoader.